_posts/2017-11-08-cost-of-drupal-7.md

```diff
@@ -0,0 +1,180 @@
+
+#### Javascript
+
+The Drupal community has been abuzz with "headless" or "decoupled" Drupal since the advent of Drupal 8. The basic idea is that Drupal can lean on its strength of being an excellent tool for highly structured and organized data in the backend while allowing freedom and flexibility of choice on the presentation layer (front end). Though [discussed two years ago](https://dri.es/should-we-decouple-drupal-with-a-client-side-framework) to no formal conclusion, Dries has recently [cited React](https://dri.es/drupal-looking-to-adopt-react) as the go-to presentation layer for administrative interfaces Drupal come early 2018. This is a fairly big deal, and formally moving forward to more tightly link with React has many implications we haven't yet fully explored. As the lines between websites and web applications continue to blur, this addition to the proudly found elsewhere may prove to be a powerful one that will not be possible for Drupal 6 or 7. We see this as another wise move to be more in-sync with the rapidly growing impact of JS frameworks.
+
+#### Access to complementary tools
+
+The re-architecture towards object-orientation helped Drupal join the modern PHP community's framework ([Cake](https://cakephp.org/), [Phalcon](https://phalconphp.com/en/), [Zend](https://framework.zend.com/), [Slim](https://www.slimframework.com/), [Laravel](https://laravel.com/), [CodeIgniter](https://www.codeigniter.com/), [Symfony](https://symfony.com/), [Yii](http://www.yiiframework.com/), and [Fuel](https://fuelphp.com/) to name the most popular) development practices. One subtle, yet substantial value to that move is now many tools that are built to help support these other frameworks are available to Drupal as well. As the toolkit for modern web development grows richer and more robust, the more Drupal can utilize, the better. A couple examples we've recently used to help inform project quality and future maintenance costs are [Code Climate](https://codeclimate.com/) and [Scrutinizer](https://scrutinizer-ci.com/). These tools have little or no value analyzing a Drupal 6 or 7 site.
+
+## Our advice
+
+So, we've dumped a lot of information on you at this point, but it may still not be entirely clear what you should do with your outdated Drupal site. Ahead we provide general suggestions as well as what is pertinent for each version separately (Drupal 6 and 7) to site owners.
+
+### To everyone on Drupal <8
```

Comment:
I didn't understand that this meant Drupal versions less than 8 rather than Drupal 8 itself until I was finished with the section

Reply:
Although maybe I'm just reading it weird because we previously talked about the whole <8 thing?